espnet2/speechlm/core_lm/builtin.py

```diff
@@ -0,0 +1,320 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Jinchuan Tian
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+from typing import Tuple, Dict
+from packaging import version
+
+import math
+import torch
+
+# (Jinchuan) TODO:
+# remove the dependency outside espnet2.speechlm for better model transplant
```

Comment:
Are there any special need you are considering over the existing modules? I thought for basic module (e.g., MHA), we can simply use the existing ones.